src/librustdoc/html/static/main.js

```diff
@@ -526,20 +526,47 @@
                 return literalSearch === true ? false : lev_distance;
             }
 
-            function checkPath(startsWith, ty) {
-                var lev_total = 0;
+            function checkPath(startsWith, lastElem, ty) {
+                var ret_lev = MAX_LEV_DISTANCE + 1;
                 var path = ty.path.split("::");
+
+                if (ty.parent && ty.parent.name) {
+                    path.push(ty.parent.name.toLowerCase());
+                }
+
                 if (startsWith.length > path.length) {
                     return MAX_LEV_DISTANCE + 1;
                 }
-                for (var i = path.length - 1; i < startsWith.length; ++i) {
-                    var lev = levenshtein(startsWith[i], path[i]);
-                    if (lev > MAX_LEV_DISTANCE) {
-                        return MAX_LEV_DISTANCE + 1;
+                for (var i = 0; i < path.length; ++i) {
+                    if (i + startsWith.length > path.length) {
+                        break;
+                    }
+                    var lev_total = 0;
+                    var aborted = false;
+                    for (var x = 0; x < startsWith.length; ++x) {
+                        var lev = levenshtein(path[i + x], startsWith[x]);
+                        if (lev > MAX_LEV_DISTANCE) {
+                            aborted = true;
+                            break;
+                        }
+                        lev_total += lev;
+                    }
+                    if (aborted === false) {
```

Comment:
Isn't it more common to check it like this: `!aborted`

Reply:
It's slower. '!' tests (from memory): `length !== 0`, `0`, `null`, `undefined`. So clearly, it's way better in js to make the comparison like this.

Reply:
Didn't know that. Thanks for explaining it.

Reply:
Except you can generally rely on the JIT to figure out that `aborted` can only contain a boolean.

Reply:
I did too much rust, I *love* being explicit. 😛